mobile/src/main/java/org/openhab/habdroid/model/WidgetDataSource.kt

```diff
@@ -41,8 +41,13 @@ class WidgetDataSource() {
             .filter { w -> w.parentId == null }
             .map { w -> w.id }
             .toSet()
-        return allWidgets
-            .filter { w -> w.parentId == null || w.parentId in firstLevelWidgetIds }
+        val secondLevelWidgetIds = allWidgets
+            .filter { w -> w.parentId in firstLevelWidgetIds && w.type in ALLOWED_SECOND_LEVEL_PARENTS }
+            .map { w -> w.id }
+            .toSet()
+        return allWidgets.filter { w ->
+            w.parentId == null || w.parentId in (firstLevelWidgetIds + secondLevelWidgetIds)
```

Comment:
I'd suggest two `in` checks instead of the concatenation to avoid the list reallocation.

Reply:
changed as suggested